### PR TITLE
Undo votes

### PR DESCRIPTION
**Pull request by rossvz**
* Restyled upvote svg with inline svg using color fill to indicate previous votes.

* Clicking on a previous vote should remove that vote

* Votes are still persisted and toggled in localStorage

### CODE REVIEW

File: src/components/Song/Upvote.js [hunk 1]
```diff
@@ -1,26 +1,23 @@
...
+    arrowStyles: {
+      width: '50px',
+      fill: isUpvoted ? 'orangered' : ''
+    }
+  }
   return (
-    <div>
-      <img alt={'vote-up'} src={Arrow} onClick={() => changeVote(1)} style={styles.arrowStyles} />
+    <div onClick={() => changeVote(isUpvoted ? -1 : 1)}
+         style={styles.arrowStyles}>
```
Comment (jhb188) [inline src/components/Song/Upvote.js]: curious whether we could move these props to the `<svg>` and get rid of the `<div>` entirely.

File: src/components/Song/Upvote.js [hunk 2]
```diff
@@ -1,26 +1,23 @@
...
 import { readVotesFromStorage } from 'util/votes'
 
-const isUpvoted = songId => {
-  let persistedVotes = readVotesFromStorage()
-  return contains(songId, persistedVotes)
-}
-
 const Upvote = ({changeVote, songId}) => {
-  if (isUpvoted(songId)) return <div>VOTED!</div>
+  const isUpvoted = contains(songId, readVotesFromStorage())
```
Comment (jhb188) [inline src/components/Song/Upvote.js]: this logic also happens in `util/votes` - we could pull it out into another exported function and save the import of `contains` in this module

File: src/img/arrow-up.svg [hunk 3]
```diff
@@ -1 +1,3 @@
-<svg xmlns="http://www.w3.org/2000/svg"  version="1.1" x="0px" y="0px" viewBox="0 0 512 640" enable-background="new 0 0 512 512" xml:space="preserve"><g><polygon points="128,320 256,192 384,320  "/></g></svg>
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 640">
+  <path d="M128 320l128-128 128 128"/>
+</svg>
```
Comment (jhb188) [inline src/img/arrow-up.svg]: do we still need this?

File: src/store/songs/actions.js [hunk 4]
```diff
@@ -7,7 +7,7 @@ export const changeVote = song => (dispatch, getState, getFirebase) =>
     const votes = song.votes || 0
     const newVotes = votes + value
     try {
-      writeVoteToStorage(song.id)
+      toggleVoteInStorage(song.id)
```
Comment (jhb188) [inline src/store/songs/actions.js]: awesome! we should call this in the `catch` of this action as well.

File: src/util/votes.js [hunk 5]
```diff
@@ -1,6 +1,21 @@
...
+}
+
+const writeVoteToStorage = songId => {
+  const persistedVotes = [ ...readVotesFromStorage(), songId ]
+  localStorage.setItem('persistedVotes', JSON.stringify(persistedVotes))
+}
+
+const removeVoteFromStorage = songId => {
+  const persistedVotes = readVotesFromStorage().filter(persistedSongId => persistedSongId !== songId)
+  debugger
```
Comment (jhb188) [inline src/util/votes.js]: do we still need these `debugger` statements?

File: src/img/arrow-up.svg (hunk 3 above)
Comment (rossvz) [reply src/img/arrow-up.svg]: nope, removed now

File: src/components/Song/Upvote.js (hunk 1 above)
Comment (rossvz) [reply src/components/Song/Upvote.js]: Yeah.. for some reason it threw an error when I had the old svg format and I had moved these into the div to try to fix. Works fine on the svg now though

File: src/components/Song/Upvote.js (hunk 2 above)
Comment (rossvz) [reply src/components/Song/Upvote.js]: Nice

File: src/util/votes.js (hunk 5 above)
Comment (rossvz) [reply src/util/votes.js]: i was just testing you

**Pull request merged**